src/operator/output.rs

```diff
@@ -0,0 +1,311 @@
+    /// [`DBSPHandle::step`](`crate::DBSPHandle::step`)
+    /// calls to retrieve the value written to the stream during the last
+    /// clock cycle.  The first call is guaranteed to return a value
+    /// (since a synchronous circuit writes exactly one value to each
+    /// stream at every clock cycle).  It removes the value from the
+    /// mailbox, so subsequent calls will return `None`.
+    ///
+    /// Invoking this method in the middle of a clock cycle, i.e., during
+    /// the execution of `DBSPHandle::step`, may return either `None` or
+    /// `Some`, depending of whether a value has already been pushed to
+    /// the stream.
+    pub fn take_from_worker(&self, worker: usize) -> Option<T> {
+        self.0.take_from_worker(worker)
+    }
+
+    /// Read values produced by all worker threads during the last
+    /// clock cycle.
+    ///
+    /// This is equivalent to calling
+    /// [`take_from_worker`](`Self::take_from_worker`) for each
+    /// worker thread in order and storing all none-`None`
+    /// results in a vector.
+    pub fn take_from_all(&self) -> Vec<T> {
+        let num_workers = self.0.mailbox.len();
+        let mut res = Vec::with_capacity(num_workers);
+
+        for worker in 0..num_workers {
+            if let Some(v) = self.take_from_worker(worker) {
+                res.push(v);
+            }
+        }
+        res
+    }
+}
+
+impl<T> OutputHandle<T>
+where
+    T: Batch<Time = ()> + Send,
+{
+    /// Read batches produced by all worker threads during the last
+    /// clock cycle and consolidate them into a single batch.
+    ///
+    /// This method is used in the common case when the `OutputHandle` is
+    /// attached to a stream that carries [`Batch`](`crate::trace::Batch`)es
+    /// of updates to relational data.  Semantically, each `Batch` consists
+    /// of `(key, value, weight)` tuples.  Depending on the structure of the
+    /// circuit, the same `key` or `(key, value)` pair can occur in batches
+    /// produced by multiple workers.  This method retrieves batches
+    /// produced by all workers and consolidates them into a single batch
+    /// where each `(key, value)` pair occurs exactly once.
+    ///
+    /// Internally, `consolidate` calls `take_from_worker` to retrieve batches
+    /// from individual worker threads.  See
+    /// [`take_from_worker`](`Self::take_from_worker`) documentation for the
+    /// exact semantics of this method.  In particular, note that repeated calls
+    /// to `take_from_worker` return `None`. `consolidate` skips `None` results
+    /// when computing the consolidated batch.
+    pub fn consolidate(&self) -> T {
+        let batches = self.take_from_all();
+        let mut spine = Spine::new(None);
+
+        for batch in batches {
+            spine.insert(batch);
+        }
+
+        spine.consolidate().unwrap_or_else(|| T::empty(()))
+    }
+}
+
+/// Sink operator that stores the contents of its input stream in
+/// the `OutputHandle`.
```

Comment:
there is no outputhandle here. Which "the"?

Reply:
The one created in `Output::new()`, I assume. Docs could be a little more clear here, I agree with that